frontend/src/components/App/PluginSettings/PluginSettings.tsx

```diff
@@ -92,20 +93,17 @@ const EnableSwitch = (props: SwitchProps) => {
 
 /** PluginSettingsPure is the main component to where we render the plugin data. */
 export function PluginSettingsPure(props: PluginSettingsPureProps) {
-  const { t } = useTranslation(['translation']);
-
-  /** Plugin arr to be rendered to the page from prop data */
-  const pluginArr: any = props.plugins ? props.plugins : [];
+  const dispatch = useDispatch();
 
-  /** enableSave state enables the save button when changes are made to the plugin list */
-  const [enableSave, setEnableSave] = useState(false);
+  const { t } = useTranslation(['translation']);
 
-  /**
-   * pluginChanges state is the array of plugin data and any current changes made by the user to a plugin's "Enable" field via toggler.
-   * The name and origin fields are split for consistency.
+  /** Plugin arr to be rendered to the page from prop data
+   * Note:
+   * The pluginArr is the main array that will be used to render the plugin data to the page.
+   * Any changes made here is what will be rendered to the page.
```

Comment:
this comment repeats the same thing three times :D

Reply:
Is the wording too non descriptive? This is kind of the most important part of the settings list so I want to make sure its known that this is the main array of the component but what is being rendered is just the copy of the pluginData prop

I was trying not to make it too lengthy but I think it needs this context to be a starting point for anyone looking at this for the first time (I know I had some old knowledge from settings it up that I had to relearn) , I'm trying to communicate that there are three things about this array part, that it:

- comes from the prop data so its a copy and not the original from the state
- this is the main array the component uses both to render the info to the menu but also for the switchChangeHandler
- any changes we make to the array here (like how we split the name for plugins with the @ symbol or changing of the enable switch) are what will be rendered and isnt entirely a 1 to 1 in the original 

changing it to something like 
```  /** 
   * Plugin array derived from prop data (copy, not original state).
   * This array is the main array used for rendering plugin info and handling switch changes.
   * Modifications done here (e.g., splitting names, toggling enable switch) are what will be rendered.
   */

Reply:
How about `// Copy of plugins with pending changes`? The fact that the state is used for rendering is not necessary to point out since pretty much all react state is used for displaying things

Reply:
done!